src/hotspot/share/runtime/javaThread.cpp

```diff
@@ -2072,3 +2076,22 @@ void JavaThread::vm_exit_on_osthread_failure(JavaThread* thread) {
                                   os::native_thread_creation_failed_msg());
   }
 }
+
+void JavaThread::pretouch_stack() {
+  // Given an established java thread stack with usable area followed by
+  // shadow zone and reserved/yellow/red zone, pretouch the usable area ranging
+  // from the current frame down to the start of the shadow zone.
+  const address end = _stack_overflow_state.shadow_zone_safe_limit();
+  if (is_in_full_stack(end)) {
```

Comment:
Shouldn't we complain to the user here if false?

Reply:
Possibly, yes. Maybe even an assert.